test/tools/yulopti.cpp

```diff
@@ -87,37 +89,70 @@ class YulOpti
 		}.printErrorInformation(_errors);
 	}
 
-	void parse(string const& _input)
+	shared_ptr<Object> getSubObject(shared_ptr<Object> const& _object, string const& _qualifiedPath)
+	{
+		if (_qualifiedPath.empty() || _qualifiedPath == _object->name.str())
+			return _object;
+
+		yulAssert(
+			boost::algorithm::starts_with(_qualifiedPath, _object->name.str() + "."),
+			"Assembly object not found."
+		);
+
+		string const subObjectPath = _qualifiedPath.substr(_object->name.str().length() + 1);
+		string const subObjectName = subObjectPath.substr(0, subObjectPath.find_first_of('.'));
+
+		auto subObjectIt = ranges::find_if(
+			_object->subObjects,
+			[&subObjectName](auto const& _subObject) { return _subObject->name.str() == subObjectName; }
+		);
+
+		yulAssert(
+			subObjectIt != _object->subObjects.end(),
+			"Assembly object not found."
+		);
+
+		auto subObject = dynamic_pointer_cast<Object>(*subObjectIt);
+
+		yulAssert(
+			subObject != nullptr,
+			"Assembly object may not contain code."
```

Comment:
Currently it sounds as if it's not allowed to contain code.
```suggestion
			"Assembly object does not contain code."
```